packages/aws-event-stream/src/aws_event_stream/events.py

```diff
@@ -387,7 +387,7 @@ def get_result(self) -> bytes:
             raise InvalidHeadersLength(len(result))
         return result
 
-    def encode_headers(self, headers: HEADERS_DICT):
+    def encode_headers(self, headers: HEADERS_DICT) -> None:
```

Comment:
I understand why this method builds an internal buffer - but the interface does seem weird.  It seems like we could either:
1. have this as a class method which would construct an instance, call _encode_headers and then return the result. or
2. Have a private class that handles the build/state, _EventHeaderEncoder which encode_headers would instantiate and return the result from.

Reply:
Yeah, I'd generally agree with this, the encode workflow here is a bit odd. I think we can look at a refactor as follow up, this PR is just fixing the typing issue from the original.

Reply:
Yeah this was intended for streaming out without an intermediary. Then we switched to have an intermediary